mithril-common/src/crypto_helper/merkle_map.rs

```diff
@@ -480,6 +491,31 @@ mod tests {
         MKTree::new(&leaves).unwrap()
     }
 
+    fn join_merkle_tree(block_ranges: &[BlockRange]) -> Vec<(BlockRange, MKTree)> {
```

Comment:
The naming of this function is not very clear:
```suggestion
    fn generate_merkle_trees_for_ranges(block_ranges: &[BlockRange]) -> Vec<(BlockRange, MKTree)> {
```